src/test/java/com/softserve/teamproject/service/impl/GroupServiceImplTest.java

```diff
@@ -0,0 +1,453 @@
+  private GroupServiceImpl groupService;
+  @Autowired
+  private GroupRepository groupRepository;
+  @Autowired
+  private LocationRepository locationRepository;
+  @Autowired
+  private EntityManager entityManager;
+  @Autowired
+  private StatusRepository statusRepository;
+  @Autowired
+  private UserRepository userRepository;
+
+  @Before
+  public void setup() {
+    MockitoAnnotations.initMocks(this);
+    doNothing().when(groupResourceAssembler).initLinks(any(), any());
+    groupService.setGroupResourceAssembler(groupResourceAssembler);
+  }
+
+  @TestGroup
+  @WithUserDetails(TEACHER_WITHOUT_GROUPS)
+  @Test
+  public void getGroupById_notExistGroupId_ReturnNull() {
+    //Arrange
+    final int GROUP_ID = 28;
+    //Act
+    Group group = groupService.getGroupById(GROUP_ID);
+    //Assert
+    assertNull(group);
+  }
+
+  @TestGroup
+  @WithUserDetails(TEACHER_WITHOUT_GROUPS)
+  @Test
+  public void getGroupResourceById_notExistGroupId_ReturnNull() {
+    //Arrange
+    final int GROUP_ID = 28;
+    //Act
+    GroupResource group = groupService.getGroupResourceById(GROUP_ID);
+    //Assert
+    assertNull(group);
+  }
+
+  @TestGroup
+  @Test
+  public void fieldsCheck() {
+    //Arrange
+    final String EXPECTED_NAME = "DP-115";
+    Group group = groupRepository.findOne(1);
+    entityManager.detach(group);
+    group.setName(null);
+    //Act
+    groupService.fieldsCheck(group);
+    //Assert
+    assertEquals(EXPECTED_NAME, group.getName());
+  }
+
+  @TestGroup
+  @WithUserDetails(TEACHER_WITHOUT_GROUPS)
+  @Test
+  public void getGroupById_existGroupId_ReturnGroup() {
+    //Arrange
+    final int GROUP_ID = 1;
+    final String EXPECTED_GROUP_NAME = "DP-115";
+    //Act
+    Group group = groupService.getGroupById(GROUP_ID);
+    //Assert
+    assertEquals(EXPECTED_GROUP_NAME, group.getName());
+  }
+
+  @TestGroup
+  @WithUserDetails(TEACHER_WITHOUT_GROUPS)
+  @Test
+  public void getGroupResourceById_existGroupId_ReturnGroup() {
+    //Arrange
+    final int GROUP_ID = 1;
+    final String EXPECTED_GROUP_NAME = "DP-115";
+    //Act
+    Group group = groupService.getGroupById(GROUP_ID);
+    //Assert
+    assertEquals(EXPECTED_GROUP_NAME, group.getName());
+  }
+
+  @TestGroup
+  @WithUserDetails(TEACHER_WITHOUT_GROUPS)
+  @Test
+  public void getAllGroupResources_teacher_returnAllGroupResources() {
+    //Arrange
+    final int EXPECTED_GROUP_COUNT = 3;
+    //Act
+    List<GroupResource> list = groupService.getAllGroupResources();
+    //Assert
+    assertEquals(EXPECTED_GROUP_COUNT, list.size());
+  }
+
+  @WithUserDetails(COORDINATOR)
+  @TestGroup
+  @Test
+  public void deleteGroup_CoordinatorHisLocationPlannedGroup_DeleteGroup() {
+    //Arrange
+    final int DELETED_GROUP_ID = 1;
+    //Act
+    groupService.deleteGroup(DELETED_GROUP_ID, COORDINATOR);
+    Group deleted = groupRepository.findOne(DELETED_GROUP_ID);
+    //Assert
+    assertNull(deleted);
+  }
+
+  @WithUserDetails(COORDINATOR)
+  @TestGroup
+  @Test
```

Comment:
sometimes i mark test annotation by exception class. so i deside do not include test into the grouptest annotation. Am I wrong? If you think that @cleartest is a more appropriate name - I chage it